File: src/main/java/com/harmonygames/harmonyengine/object/component/Component.java
package com.harmonygames.harmonyengine.object.component;

import com.harmonygames.harmonyengine.object.GameObject;

public abstract class Component {

    protected GameObject gameObject = null;

    public abstract void onCreate();
    public abstract void update(float deltaTime);

    public void setGameObject(GameObject gameObject) { this.gameObject = gameObject; }
    public GameObject getGameObject() { return this.gameObject; }

    public boolean isSharable() { return false; }
}

File: HarmonyEngine/src/Scene/Component.cpp
#include "Component.h"

#include "GameObject.h"

// ======================================================================================
// Component Class
// ======================================================================================
void Component::OnCreate() {}
void Component::Update(const float& deltaTime) {}
void Component::OnDestroy() {}

GameObject* Component::GetParentGameObject() { return m_ParentObject; }

// ======================================================================================
// Quad Renderer Componet
// ======================================================================================
void QuadRenderer::OnCreate() {}

void QuadRenderer::Update(const float& deltaTime) {
    Log::Info("Hello World!");
    Renderer::DrawQuad(m_Quad);
}

void QuadRenderer::OnDestroy() {}